### ISSUE TITLE
How to handle unknown zone errors

### ISSUE THREAD

**Issue by milesj** [moment/luxon#311]
We're seeing a few errors in Sentry for "Invalid DateTime: unsupported zone". First, it's hard to tell what the zone is (perhaps it can be in the error message). Second, how to resolve this? Or fallback to UTC or something?

**Comment by icambron** [comment]
Yeah, it should be in the message. Would accept a PR setting that.

As for how to handle it, that depends on the use case. What are you using the zone for?

**Comment by milesj** [comment]
We have a helper function that creates a `DateTime` object, and automatically handles our apps locale/timezone settings. The TZ defaults by using `Intl` (similar to `moment.tz.guess()`). I guess whatever value that returns is not valid.

**Comment by icambron** [comment]
You might consider just defaulting to local time instead of manually setting the zone to the system zone. Should work the same without this pitfall.

FWIW, the zone passed in is also parsed by Intl, and there are at least a few places where there's some asymmetry between what it produces and what it can parse in some browsers. Here's an example from Chrome: https://bugs.chromium.org/p/chromium/issues/detail?id=364374&q=gmt%20intl&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified

**Comment by milesj** [comment]
Thank you, I'll keep that in mind!

**Comment by mixth** [comment]
Hi. I'm new to the open source world and this might be my first contribution. :)

What do you have in mind for displaying the invalid zone? Could it be:
```
Invalid DateTime: unsupported zone 'blorp'
```
When 'blorp' is the zone given.

**Comment by milesj** [comment]
Yup, something as simple as that would be great.

**Comment by icambron** [comment]
See #304. I'll add some comments there

**Comment by icambron** [comment]
This is fixed in #360, where you can get

```js
DateTime.local().setZone("America/Blorp").invalidExplanation
'the zone "America/Blorp" is not supported'
```

Will be in 1.5.0.

**Comment by milesj** [comment]
Awesome!
